### PR TITLE
[MobileNetwork] Fixissue mobile network logic for New-Az*

### PR DESCRIPTION
**Pull request by lijinpei2008**
## Description
Issue: https://github.com/Azure/autorest.powershell/issues/1114

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/MobileNetwork/custom/Update-AzMobileNetworkDataNetwork.ps1 [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    [Parameter(DontShow)]
+    [Microsoft.Azure.PowerShell.Cmdlets.MobileNetwork.Category('Runtime')]
+    [System.Management.Automation.SwitchParameter]
+    # Use the default credentials for the proxy
+    ${ProxyUseDefaultCredentials}
+)
+
+    process {
+        try {
+            $dataBase = Get-AzMobileNetworkDataNetwork -MobileNetworkName $PSBoundParameters.MobileNetworkName -ResourceGroupName $PSBoundParameters.ResourceGroupName -Name $PSBoundParameters.DataNetworkName
```
Comment (wyunchi-ms) [inline src/MobileNetwork/custom/Update-AzMobileNetworkDataNetwork.ps1:147]: The variable should named as `dataNetwork`.

File: src/MobileNetwork/custom/Update-AzMobileNetworkAttachedDataNetwork.ps1 [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+    [Parameter(DontShow)]
+    [Microsoft.Azure.PowerShell.Cmdlets.MobileNetwork.Category('Runtime')]
+    [System.Management.Automation.SwitchParameter]
+    # Use the default credentials for the proxy
+    ${ProxyUseDefaultCredentials}
+)
+
+    process {
+        try {
+            $dataBase = Get-AzMobileNetworkAttachedDataNetwork -PacketCoreControlPlaneName $PSBoundParameters.PacketCoreControlPlaneName -PacketCoreDataPlaneName $PSBoundParameters.PacketCoreDataPlaneName -ResourceGroupName $PSBoundParameters.ResourceGroupName -Name $PSBoundParameters.AttachedDataNetworkName
```
Comment (wyunchi-ms) [inline src/MobileNetwork/custom/Update-AzMobileNetworkAttachedDataNetwork.ps1:260]: The variable should be named as `attachedDataNetwork`

File: src/MobileNetwork/custom/Update-AzMobileNetworkPacketCoreDataPlane.ps1 [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    [Parameter(DontShow)]
+    [Microsoft.Azure.PowerShell.Cmdlets.MobileNetwork.Category('Runtime')]
+    [System.Management.Automation.SwitchParameter]
+    # Use the default credentials for the proxy
+    ${ProxyUseDefaultCredentials}
+)
+
+    process {
+        try {
+            $dataBase = Get-AzMobileNetworkPacketCoreDataPlane -PacketCoreControlPlaneName $PSBoundParameters.PacketCoreControlPlaneName -ResourceGroupName $PSBoundParameters.ResourceGroupName -Name $PSBoundParameters.PacketCoreDataPlaneName
```
Comment (wyunchi-ms) [inline src/MobileNetwork/custom/Update-AzMobileNetworkPacketCoreDataPlane.ps1:166]: The variable should named as `packetCoreDataPlane`.

File: src/MobileNetwork/custom/Update-AzMobileNetworkSimGroup.ps1 [hunk 4]
```diff
@@ -154,47 +159,38 @@ param(
...
-        if ($PSBoundParameters.TryGetValue('OutBuffer', [ref]$outBuffer)) {
-            $PSBoundParameters['OutBuffer'] = 1
-        }
-        $parameterSet = $PSCmdlet.ParameterSetName
-
-        $mapping = @{
-            CreateExpanded = 'Az.MobileNetwork.private\New-AzMobileNetworkSimGroup_CreateExpanded';
+    process {
+        try {
+            $dataBase = Get-AzMobileNetworkSimGroup -ResourceGroupName $PSBoundParameters.ResourceGroupName -Name $PSBoundParameters.SimGroupName
```
Comment (wyunchi-ms) [inline src/MobileNetwork/custom/Update-AzMobileNetworkSimGroup.ps1:164]: The variable should named as `simGroup`.

File: src/MobileNetwork/custom/Update-AzMobileNetworkSimPolicy.ps1 [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+    [Parameter(DontShow)]
+    [Microsoft.Azure.PowerShell.Cmdlets.MobileNetwork.Category('Runtime')]
+    [System.Management.Automation.SwitchParameter]
+    # Use the default credentials for the proxy
+    ${ProxyUseDefaultCredentials}
+)
+
+    process {
+        try {
+            $dataBase = Get-AzMobileNetworkSimPolicy -MobileNetworkName $PSBoundParameters.MobileNetworkName -ResourceGroupName $PSBoundParameters.ResourceGroupName -Name $PSBoundParameters.SimPolicyName
```
Comment (wyunchi-ms) [inline src/MobileNetwork/custom/Update-AzMobileNetworkSimPolicy.ps1:181]: The variable should named as `simPolicy`.

File: src/MobileNetwork/custom/Update-AzMobileNetworkDataNetwork.ps1 (hunk 1 above)
Comment (lijinpei2008) [reply src/MobileNetwork/custom/Update-AzMobileNetworkDataNetwork.ps1]: Done.

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**